content/projects/2022-mixed-precision/index.md

```diff
@@ -7,3 +7,148 @@ weight : 100
+$$\min_{x} \{||Ax-b||_2^2+\lambda^2||x||_2^2\}$$ where $\lambda$ is generated by the function [IRhybrid_lsqr](https://www.emis.de/journals/ETNA/vol.28.2007-2008/pp149-167.dir/pp149-167.pdf).
+
+From the graph below, it is clear that even with 10% noise, the half graph looks very similar to that in double precision, better than what we have using CGLS. 
+<p align="center">
+<img src="img/cs_reg_0.1_blur.png" alt="draw" width="500"/> 
+</p>
+For image deblurring problem, We further comfirmed by plotting the error norm. 
+<p align="center">
+<img src="img/cs_reg_0.1_blur_Enrm.png" alt="draw" width="300"/> 
+</p>
+We can see that the error norms of three precisions overlap, proving the result in half precision is similar to that in double precision. 
+
+### Tomography Reconstruction Using CS
+We also ran the code to solve tomography reconstruction too, and the result is showed below. 
+<p align="center">
+<img src="img/cs_reg_0.1_tomo.png" alt="draw" width="500"/> 
+</p>
+Although there is no inner product in CS, we still have overflow in half precision, which is because the A is too large. Therefore, we rescale A and b by diving both of by 100, so that the algorithm successfully ran to the end without generating NaNs. The image in half precision is as good as double precision, possessing clear boundaries and backgrounds. The error norm also shows great overlappings among the three precisions.
+
+### Image Deblurring Using CGLS with regularizationand 
+To fully compare CGLS and CS, we added Tikhonov regularization to CGLS and ran the two test problems. The diagrams are listed below.
+We also ran the code to solve tomography reconstruction too, and the result is showed below. 
+<p align="center">
+<img src="img/cg_reg_blur_64_0.1_m.png" alt="draw" width="500"/> 
+</p>
+<p align="center">
+<img src="img/cg_reg_m_0.1_blur_Enrm.png" alt="draw" width="300"/> 
+</p>
+The diagram for half precision looks much better than that produced by CGLS without regularization. However, in the background there is still a different between half and double precision. From the error norm, we can see that at the end of the iteration for half precision, the error norm still starts to go up. To see the error norms more clearly, we zoomed in the picture of error norm for CS and CGLS with regularization, and this is what we got:
+<p align="center">
+<img src="img/Zoom in of Enrm for cs and cg.png" alt="draw" width="500"/> 
+</p>
+We can see that the error norm for half precision converges for CS but increases rapidly for CGLS, suggesting that for half precision, CS is a better choice. However, for double precision, the CGLS with regularization is clearly more stable. Therefore, CGLS with regularizatin is more suitable for double precision. The result is similar For tomography reconstruction to this, so it not shown here. There is detailed information in our [paper](), and we welcome you to check it out.
+
```

Comment:
Think about how you want to end the article? Is there a conclusion/summary of the experiment? Or do you want to add pictures and short bios?